[sui-execution][EASY] README (#12542)
MystenLabs/sui: sui-execution/cut/README.md

# Sui Execution: Cut

The `cut` binary is responsible for making copies of sets of crates
from the repository.  Use the `--dry-run` option to inspect the copies
that `cut` would make without committing them to the filesystem.

## Properties

When making copies, `cut` maintains the following properties:

- If crate A depends on crate B, then when copies A' and B' are made,
  A' will depend on B' (not B).
- If crate A (copied) depends on crate B (not copied) and refers to it

>   by a relative path, the path in copy A' is fixed up to continue

Reviewer: ```suggestion
  by a relative path, the path in copy A' continues
```

>   pointing to B

Reviewer: ```suggestion
  pointing to B.
```
:)

- If crate C is a workspace member, then the copy, C', will also be
  added to the workspace members.
- If crate C is a workspace exclude, then the copy, C', will also be
  added to the workspace excludes.

## Finding Crates

The binary accepts two sets of configuration for discovering packages
to copy:

- A set of directories to search for crates, along with a destination
  directory to put copies of crates found within this directory, and
  an optional `suffix` parameter that is used to strip a common suffix
  from the names of crates found in this directory.
- A list of crate names to look for -- only crates whose names are in
  this list will be copied, and if a crate in this list is not
  encountered, a warning will be issued.

## Copying

When copying crates, the tool preserves the relative path of the crate

> in its source directory. I.e. if crate A is found at path

Reviewer: ```suggestion
in its source directory. For instance, if crate A is found at path
```
